### ISSUE TITLE
Minimize MacPass to menu bar icon

### ISSUE THREAD

**Issue by DevJackSmith** [mstarke/MacPass#484]
Hello and thanks for a great app.

I'd like to request a feature, if it's not too much hassle:

As a user I'd like to be able to have the app in the menu bar icon up top only.  On app start and when minimizing/closing the app.

Thanks again.

**Comment by cornercase** [comment]
I am part way through implementing this feature.  I have added a menu bar icon and can selectively enable the menu bar and dock icons. @mstarke, feel free to assign this to me if you would like.  I'll hopefully have a merge request by early next week.  If there are any inputs on design or technical implementation details, let me know.

I merge the proposed menu bar icon into the MacPassHTTP icon? 

### Proposed UI Changes ###
* Add 2 check boxes to the General Settings Windows, both enabled by default
 * Show Dock Icon
 * Show Menu Bar Icon
* When user clicks menu bar icon...
 * Show MacPass window if hidden
 * Hide MacPass if shown

**Comment by ericmoret** [comment]
It would be nice to add an option to unlock MacPass directly from the Menu Bar Icon.

**Comment by leksuss** [comment]
It would be great if this feature will be implemented.

**Comment by Alph2** [comment]
I couldn't agree more : would be a great feature. @cornercase Did you give up on adding it?

**Comment by mstarke** [comment]
The core functionality is rather straight forward, but I do not know the impact it hast when interacting with MacPass' windows when the Dock icon is hidden. Have to run some tests. But I should be able to add this rather quickly. @cornercase seems to not have gone through with his changes or might have chosen to keep them private :)

**Comment by cornercase** [comment]
I started working on the changes and then ran into a few roadblocks and got distracted by writing my thesis.  I am hoping to get back to the changes in a few weeks now that my life is starting to calm down a little bit, will keep you posted.


On October 23, 2017 at 2:14:22 AM, Michael Starke (notifications@github.com) wrote:

The core functionality is rather straight forward, but I do not know the impact it hast when interacting with MacPass' windows when the Dock icon is hidden. Have to run some tests. But I should be able to add this rather quickly. @cornercase seems to not have gone through with his changes or might have chosen to keep them private :)

**Comment by georgesnow** [comment]
yeah this would be great feature. as the first post mentioned 1Password has global "Show 1P" its keyboard but it also the same as selecting the menu bar icon. which would be nice. 

possibly have the focus should be the search field. that would make it easy to type for the entry you want. Then one step further you can trigger auto type for entry or just double click to copy password, which would hide MacPass.

Example would be i connect to a SMB share or SFTP in Fetch/Cyberduck. 
Password requested. Select the pw field. Use global keyboard shortcut or menubar icon to reveal MacPass. Focus lands in search. Find Entry. Trigger Autotype. 

This resolves applications windows where the window doesn't have a name "Global Autotype" can match easily. (ex. "item-0" or weird ones like "Dock Progress" which i often get sadly :( RemoteDesktop and RemoteApps, iTerm, Fetch etc..) It also solves if you have multiple entries for window that can't be easily matched application or window. It makes it truly agnostic to the application at point.

**Comment by georgesnow** [comment]
i built a plugin that provides part of this functionality (and user configured global hotkey to hide/show). if you are interested. take a look at 
Duplicate of #941 

I am still working on hiding the Dock icon. There is some weird UX issues when the plugin hides the dock icon I haven't worked out. But if the code is merged into the core code the hiding of the dock icon might resolve itself.

**Comment by orestesgaolin** [comment]
Implemented https://github.com/orestesgaolin/MacPass/pull/4
